internal/network/manifests_generator.go

```diff
@@ -492,3 +494,94 @@ func NewConfig() (*Config, error) {
 	}
 	return &networkCfg, nil
 }
+
+const nodeIpHint = `
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: {{.ROLE}}
+  name: 10-{{.ROLE}}s-node-ip-hint
+spec:
+  config:
+    ignition:
+      version: 3.1.0
+    storage:
+      files:
+        - contents:
+            source: data:text/plain;charset=utf-8;base64,{{.NODE_IP_CONTENT}}
+            verification: {}
+          filesystem: root
+          mode: 420
+          path: /etc/default/nodeip-configuration
+`
+
+// Add node ip hint (is supported from 4.10 but it makes no harm to push this file to any version)
+// it will allow us to tell to node-ip script which ip kubelet should run with
+// https://github.com/openshift/machine-config-operator/commit/a0c9a3caa54018eb89eb5bdd6ec1b8fbf97f6fb7
+func (m *ManifestsGenerator) AddNodeIpHint(ctx context.Context, log logrus.FieldLogger, cluster *common.Cluster) error {
+	if !common.IsSingleNodeCluster(cluster) {
+		return nil
+	}
+
+	if hintSupported, err := common.VersionGreaterOrEqual(cluster.OpenshiftVersion, "4.10.15"); err != nil || !hintSupported {
+		return err
+	}
+
+	if !IsMachineCidrAvailable(cluster) {
+		return fmt.Errorf("node-ip-hint allowed only if machine network was configured")
+	}
+
+	bootstrap := common.GetBootstrapHost(cluster)
+	inventory, err := common.UnmarshalInventory(bootstrap.Inventory)
+	if err != nil {
+		log.WithError(err).Errorf("Failed to unmarshal bootstrap inventory")
+		return err
+	}
+
+	hostNetworks, err := getHostNetworks(inventory, func(i *models.Interface) []string { return append(i.IPV4Addresses, i.IPV6Addresses...) })
+	if err != nil {
+		log.WithError(err).Errorf("Failed to get host networks")
+		return err
+	}
+	// if we have only one network there
+	// is not need to set those manifests
+	if len(hostNetworks) < 2 {
+		log.Debugf("SNO cluster has only one network, no need to add ip hint manifests")
+		return nil
+	}
+	log.Infof("Adding node add ip hint manifests")
+
+	for _, role := range []models.HostRole{models.HostRoleMaster, models.HostRoleWorker} {
```

Comment:
Why do we need the manifest for both roles?

Reply:
for day2